tests/pytorch/test_match_dimensions.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import pytest
+import torch
+
+from amltk import (
+    Component,
+    Fixed,
+    MatchDimensions,
+    Node,
+    Sequential,
+    build_model_from_pipeline,
+)
+from amltk.exceptions import MatchDimensionsError
+from tests.pytorch.common import create_optimizer
+
+
+class TestMatchDimensions:
+    @pytest.fixture(scope="class")
+    def valid_pipeline(self) -> Node:
+        """Fixture to create a valid pipeline."""
+        return Sequential(
+            Component(
+                torch.nn.Linear,
+                config={"in_features": 784},
+                space={"out_features": (64, 128)},
+                name="fc1",
+            ),
+            Fixed(torch.nn.Sigmoid(), name="activation"),
+            Component(
+                torch.nn.Linear,
+                config={
+                    "in_features": MatchDimensions("fc1", param="out_features"),
+                    "out_features": MatchDimensions("fc3", param="in_features"),
+                },
+                name="fc2",
+            ),
+            Component(
+                torch.nn.Linear,
+                space={"in_features": (128, 256)},
+                config={"out_features": 10},
+                name="fc3",
+            ),
+            Fixed(torch.nn.LogSoftmax(dim=1), name="log_softmax"),
+            name="my-pipeline",
+        )
+
+    @pytest.fixture(
+        scope="class",
+        params=[
+            MatchDimensions("non-existing-layer", param="out_features"),
+            MatchDimensions("fc1", param="non-existing-param"),
+            MatchDimensions("fc1", param=None),
+            MatchDimensions(layer_name="", param="out_features"),
+        ],
+    )
```

Comment:
Is this like a parametrized fixture?

Reply:
Yes, exactly. It runs 4 times with different invalid configuration.